Sources/Commands/SwiftTestTool.swift

```diff
@@ -983,9 +1009,12 @@ final class ParallelTestRunner {
         // Report the completion.
         progressAnimation.complete(success: processedTests.get().contains(where: { !$0.success }))
 
+        let testResults = processedTests.get()
+        let atLeastOneTestFailed = testResults.contains(where: { !$0.success })
+
         // Print test results.
-        for test in processedTests.get() {
-            if (!test.success || shouldOutputSuccess) && !buildParameters.testingParameters.experimentalTestOutput {
+        for test in testResults {
+            if (!test.success || shouldOutputSuccess || (atLeastOneTestFailed && loggingOptions.quiet)) && !buildParameters.testingParameters.experimentalTestOutput {
```

Comment:
nit: line length, in newly written or updated code we're sticking to 120 characters per [the formatting config](https://github.com/apple/swift-package-manager/blob/ace3488c153f2d99732611109408d02186b75c20/.swiftformat#L15)